gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java

```diff
@@ -115,6 +212,84 @@ public static ProcessStreamReader start(Process process, String blockUntil) thro
       return thread;
     }
   }
+  
+  private static class CommandWrapper {
+    
+    private final List<String> prefix;
+    private List<String> command;
+    private String nullFilename;
+    private boolean redirectOutputToNull;
+    private boolean redirectErrorStream;
+    private boolean redirectErrorInherit;
+    private Path directory;
+    
+    private CommandWrapper() {
+      this.prefix = new ArrayList<>();
+    }
+    
+    public CommandWrapper command(String... command) {
+      this.command = new ArrayList<>(command.length + this.prefix.size());
+      this.command.addAll(prefix);
+      this.command.addAll(Arrays.asList(command));
+      return this;
+    }
+    
+    public CommandWrapper redirectOutputToNull() {
+      this.redirectOutputToNull = true;
+      return this;
+    }
+    
+    public CommandWrapper redirectErrorStream() {
+      this.redirectErrorStream = true;
+      return this;
+    }
+    
+    public CommandWrapper redirectErrorInherit() {
+      this.redirectErrorInherit = true;
+      return this;
+    }
+    
+    public CommandWrapper directory(Path directory) {
+      this.directory = directory;
+      return this;
+    }
+    
+    public ProcessBuilder builder() {
+      ProcessBuilder builder = new ProcessBuilder(command);
+      if (redirectOutputToNull) {
+        builder.redirectOutput(new File(nullFilename));
+      }
+      if (directory != null) {
+        builder.directory(directory.toFile());
+      }
+      if (redirectErrorStream) {
+        builder.redirectErrorStream(true);
+      }
+      if (redirectErrorInherit) {
+        builder.redirectError(ProcessBuilder.Redirect.INHERIT);
+      }
+      return builder;
+    }
+    
+    public Process start() throws IOException {
+      return builder().start();
+    }
+    
+    public static CommandWrapper cmd() {
+      CommandWrapper wrapper = new CommandWrapper();
```

Comment:
I was wondering which was the best solution when I implemented it. I went for the two factory methods as a `isWindows` check was already in the caller. But you are right, it's nicer to have a single factory method.